shesha/supervisor/stage2Supervisor.py

```diff
@@ -0,0 +1,194 @@
+## @package   shesha.supervisor.stage2Supervisor
+## @brief     Initialization and execution of a second stage supervisor
+## @author    SAXO+ Team (Clementine Bechet)
+## @version   5.3.0
+## @date      2023/01/31
+## @copyright GNU Lesser General Public License
+#
+#  This file is part of COMPASS <https://anr-compass.github.io/compass/>
+#
+#  Copyright (C) 2011-2022 COMPASS Team <https://github.com/ANR-COMPASS>
+#  All rights reserved.
+#  Distributed under GNU - LGPL
+#
+#  COMPASS is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+#  General Public License as published by the Free Software Foundation, either version 3 of the License,
+#  or any later version.
+#
+#  COMPASS: End-to-end AO simulation tool using GPU acceleration
+#  The COMPASS platform was designed to meet the need of high-performance for the simulation of AO systems.
+#
+#  The final product includes a software package for simulating all the critical subcomponents of AO,
+#  particularly in the context of the ELT and a real-time core based on several control approaches,
+#  with performances consistent with its integration into an instrument. Taking advantage of the specific
+#  hardware architecture of the GPU, the COMPASS tool allows to achieve adequate execution speeds to
+#  conduct large simulation campaigns called to the ELT.
+#
+#  The COMPASS platform can be used to carry a wide variety of simulations to both testspecific components
+#  of AO of the E-ELT (such as wavefront analysis device with a pyramid or elongated Laser star), and
+#  various systems configurations such as multi-conjugate AO.
+#
+#  COMPASS is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+#  implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License along with COMPASS.
+#  If not, see <https://www.gnu.org/licenses/lgpl-3.0.txt>.
+
+from shesha.supervisor.compassSupervisor import CompassSupervisor
+from shesha.supervisor.components import AtmosCompass, DmCompass, RtcCompass, TargetCompass, TelescopeCompass, WfsCompass
+from shesha.supervisor.optimizers import ModalBasis, Calibration, ModalGains
+import numpy as np
+import time
+
+import shesha.constants as scons
+
+from typing import Iterable
+
+
+class Stage2Supervisor(CompassSupervisor):
```

Comment:
I have the feeling that Stage1Supervisor & Stage2Supervisor are really close to each other, with a lot of code duplication.
Could be better to "merge" those two classes into a single one, and to handle the runtime differences with additional function arguments...

Reply:
Addressed by [d484b28](https://github.com/ANR-COMPASS/shesha/pull/44/commits/d484b284b183f9b2c04c235e5f67392db45019de)